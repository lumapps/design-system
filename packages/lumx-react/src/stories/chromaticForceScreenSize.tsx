/* eslint-disable */
import React from 'react';
import isChromatic from 'chromatic/isChromatic';

/** Story decorator used to force a minimum screen size when running in chromatic */
export const chromaticForceScreenSize = (story: any) =>
    isChromatic() ? <div style={{ minWidth: 1200, minHeight: 800 }}>{story()}</div> : story();
