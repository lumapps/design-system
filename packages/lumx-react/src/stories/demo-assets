../../../site-demo/static/demo-assets