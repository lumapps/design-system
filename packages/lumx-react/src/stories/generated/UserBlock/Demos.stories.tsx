/**
 * File generated when storybook is started. Do not edit directly!
 */
export default { title: 'LumX components/user-block/Demos' };

export { App as Extended } from './extended';
export { App as SizeL } from './size-l';
export { App as SizeM } from './size-m';
export { App as SizeS } from './size-s';
export { App as Vertical } from './vertical';
