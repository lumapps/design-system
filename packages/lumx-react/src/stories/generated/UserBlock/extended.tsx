/home/gcornut/git/dup-design-system/packages/site-demo/content/product/components/user-block/react/extended.tsx