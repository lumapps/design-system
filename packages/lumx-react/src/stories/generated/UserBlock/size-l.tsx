/home/gcornut/git/design-system/packages/site-demo/content/product/components/user-block/react/size-l.tsx