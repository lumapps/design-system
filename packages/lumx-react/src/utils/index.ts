/**
 * WARNING: All modules exported here are exposed to NPM in '@lumx/react/utils'.
 */

export { ClickAwayProvider } from './ClickAwayProvider';
export { Portal, type PortalProps, type PortalInit, PortalProvider, type PortalProviderProps } from './Portal';
