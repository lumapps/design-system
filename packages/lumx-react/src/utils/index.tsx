export * from '@lumx/core/js/utils';

export * from './getRootClassName';

export * from './htmlDecode';

export * from './partitionMulti';

export * from './type';
