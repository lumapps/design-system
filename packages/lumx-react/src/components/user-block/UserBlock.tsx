import React, { forwardRef, ReactNode } from 'react';

import classNames from 'classnames';

import { Avatar, Orientation, Size, Theme } from '@lumx/react';

import { Comp, GenericProps, getRootClassName, handleBasicClasses } from '@lumx/react/utils';
import { AvatarProps } from '../avatar/Avatar';

/**
 * User block sizes.
 */
export type UserBlockSize = Extract<Size, 's' | 'm' | 'l'>;

/**
 * Defines the props of the component.
 */
export interface UserBlockProps extends GenericProps {
    /** Props to pass to the avatar. */
    avatarProps?: AvatarProps;
    /** Simple action toolbar content. */
    simpleAction?: ReactNode;
    /** Multiple action toolbar content. */
    multipleActions?: ReactNode;
    /** Additional fields used to describe the user. */
    fields?: string[];
    /** User name. */
    name?: string;
    /** Orientation. */
    orientation?: Orientation;
    /** Size variant. */
    size?: UserBlockSize;
    /** Theme adapting the component to light or dark background. */
    theme?: Theme;
    /** On click callback. */
    onClick?(): void;
    /** On mouse enter callback. */
    onMouseEnter?(): void;
    /** On mouse leave callback. */
    onMouseLeave?(): void;
}

/**
 * Component display name.
 */
const COMPONENT_NAME = 'UserBlock';

/**
 * Component default class name and class prefix.
 */
const CLASSNAME = getRootClassName(COMPONENT_NAME);

/**
 * Component default props.
 */
const DEFAULT_PROPS: Partial<UserBlockProps> = {
    orientation: Orientation.horizontal,
    size: Size.m,
    theme: Theme.light,
};

/**
 * UserBlock component.
 *
 * @param  props Component props.
 * @param  ref   Component ref.
 * @return React element.
 */
export const UserBlock: Comp<UserBlockProps, HTMLDivElement> = forwardRef((props, ref) => {
    const {
        avatarProps,
        className,
        fields,
        multipleActions,
        name,
        onClick,
        onMouseEnter,
        onMouseLeave,
        orientation,
        simpleAction,
        size,
        theme,
        ...forwardedProps
    } = props;
    let componentSize = size;

    // Special case - When using vertical orientation force the size to be Sizes.l.
    if (orientation === Orientation.vertical) {
        componentSize = Size.l;
    }

    const shouldDisplayActions: boolean = orientation === Orientation.vertical;

    const nameBlock: ReactNode = name && (
        // eslint-disable-next-line jsx-a11y/click-events-have-key-events,jsx-a11y/no-noninteractive-tabindex,jsx-a11y/no-static-element-interactions
        <span className={`${CLASSNAME}__name`} onClick={onClick} tabIndex={onClick ? 0 : -1}>
            {name}
        </span>
    );

    const fieldsBlock: ReactNode = fields && componentSize !== Size.s && (
        <div className={`${CLASSNAME}__fields`}>
            {fields.map((aField: string, idx: number) => (
                <span key={idx} className={`${CLASSNAME}__field`}>
                    {aField}
                </span>
            ))}
        </div>
    );

    return (
        <div
            ref={ref}
            {...forwardedProps}
            className={classNames(
                className,
                handleBasicClasses({ prefix: CLASSNAME, orientation, size: componentSize, theme }),
            )}
            onMouseLeave={onMouseLeave}
            onMouseEnter={onMouseEnter}
        >
            {avatarProps && (
                <div className={`${CLASSNAME}__avatar`}>
                    <Avatar
                        {...avatarProps}
                        size={componentSize}
                        onClick={onClick}
                        tabIndex={onClick ? 0 : -1}
                        theme={theme}
                    />
                </div>
            )}
            {(fields || name) && (
                <div className={`${CLASSNAME}__wrapper`}>
                    {nameBlock}
                    {fieldsBlock}
                </div>
            )}
            {shouldDisplayActions && simpleAction && <div className={`${CLASSNAME}__action`}>{simpleAction}</div>}
            {shouldDisplayActions && multipleActions && (
                <div className={`${CLASSNAME}__actions`}>{multipleActions}</div>
            )}
        </div>
    );
});
UserBlock.displayName = COMPONENT_NAME;
UserBlock.className = CLASSNAME;
UserBlock.defaultProps = DEFAULT_PROPS;
