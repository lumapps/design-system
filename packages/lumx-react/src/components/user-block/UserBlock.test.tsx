import React, { ReactElement } from 'react';

import { mount, shallow } from 'enzyme';
import 'jest-enzyme';

import { commonTestsSuite, itShouldRenderStories, Wrapper } from '@lumx/react/testing/utils';

import { UserBlock, UserBlockProps } from './UserBlock';
import * as stories from './UserBlock.stories';

const CLASSNAME = UserBlock.className as string;

/**
 * Mounts the component and returns common DOM elements / data needed in multiple tests further down.
 */
const setup = ({ ...propsOverride }: Partial<UserBlockProps> = {}, shallowRendering = true) => {
    const props: UserBlockProps = { ...propsOverride };
    const renderer: (el: ReactElement) => Wrapper = shallowRendering ? shallow : mount;
    const wrapper: Wrapper = renderer(<UserBlock {...props} />);

    return { props, wrapper };
};

describe(`<${UserBlock.displayName}>`, () => {
    // 1. Test render via snapshot.
    describe('Snapshots and structure', () => {
        itShouldRenderStories(stories, UserBlock);
    });

    // Common tests suite.
    commonTestsSuite(setup, { className: 'wrapper', prop: 'wrapper' }, { className: CLASSNAME });
});
