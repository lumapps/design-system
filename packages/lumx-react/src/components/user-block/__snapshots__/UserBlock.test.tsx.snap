// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserBlock> Snapshots and structure should render story 'InList' 1`] = `
Array [
  <div
    className="lumx-user-block lumx-user-block--orientation-horizontal lumx-user-block--size-m lumx-user-block--theme-light"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="lumx-user-block__avatar"
    >
      <Avatar
        alt="Avatar"
        badge={
          <Badge
            color="blue"
          >
            <Icon
              icon="M12,17.27L18.18,21L16.54,13.97L22,9.24L14.81,8.62L12,2L9.19,8.62L2,9.24L7.45,13.97L5.82,21L12,17.27Z"
            />
          </Badge>
        }
        image="avatar1.jpg"
        onClick={[Function]}
        size="m"
        tabIndex={0}
        theme="light"
      />
    </div>
    <div
      className="lumx-user-block__wrapper"
    >
      <span
        className="lumx-user-block__name"
        onClick={[Function]}
        tabIndex={0}
      >
        Emmitt O. Lum
      </span>
      <div
        className="lumx-user-block__fields"
      >
        <span
          className="lumx-user-block__field"
          key="0"
        >
          Creative developer
        </span>
        <span
          className="lumx-user-block__field"
          key="1"
        >
          Denpasar
        </span>
      </div>
    </div>
  </div>,
  <div
    className="lumx-user-block lumx-user-block--orientation-horizontal lumx-user-block--size-m lumx-user-block--theme-light"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="lumx-user-block__avatar"
    >
      <Avatar
        alt="Avatar"
        badge={
          <Badge
            color="blue"
          >
            <Icon
              icon="M12,17.27L18.18,21L16.54,13.97L22,9.24L14.81,8.62L12,2L9.19,8.62L2,9.24L7.45,13.97L5.82,21L12,17.27Z"
            />
          </Badge>
        }
        image="avatar2.jpg"
        onClick={[Function]}
        size="m"
        tabIndex={0}
        theme="light"
      />
    </div>
    <div
      className="lumx-user-block__wrapper"
    >
      <span
        className="lumx-user-block__name"
        onClick={[Function]}
        tabIndex={0}
      >
        Emmitt O. Lum
      </span>
      <div
        className="lumx-user-block__fields"
      >
        <span
          className="lumx-user-block__field"
          key="0"
        >
          Creative developer
        </span>
        <span
          className="lumx-user-block__field"
          key="1"
        >
          Denpasar
        </span>
      </div>
    </div>
  </div>,
  <div
    className="lumx-user-block lumx-user-block--orientation-horizontal lumx-user-block--size-m lumx-user-block--theme-light"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="lumx-user-block__avatar"
    >
      <Avatar
        alt="Avatar"
        badge={
          <Badge
            color="blue"
          >
            <Icon
              icon="M12,17.27L18.18,21L16.54,13.97L22,9.24L14.81,8.62L12,2L9.19,8.62L2,9.24L7.45,13.97L5.82,21L12,17.27Z"
            />
          </Badge>
        }
        image="avatar3.jpg"
        onClick={[Function]}
        size="m"
        tabIndex={0}
        theme="light"
      />
    </div>
    <div
      className="lumx-user-block__wrapper"
    >
      <span
        className="lumx-user-block__name"
        onClick={[Function]}
        tabIndex={0}
      >
        Emmitt O. Lum
      </span>
      <div
        className="lumx-user-block__fields"
      >
        <span
          className="lumx-user-block__field"
          key="0"
        >
          Creative developer
        </span>
        <span
          className="lumx-user-block__field"
          key="1"
        >
          Denpasar
        </span>
      </div>
    </div>
  </div>,
]
`;

exports[`<UserBlock> Snapshots and structure should render story 'Sizes' 1`] = `
Array [
  <div
    className="lumx-user-block lumx-user-block--orientation-horizontal lumx-user-block--size-s lumx-user-block--theme-light"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="lumx-user-block__avatar"
    >
      <Avatar
        alt="Avatar"
        image="avatar1.jpg"
        onClick={[Function]}
        size="s"
        tabIndex={0}
        theme="light"
      />
    </div>
    <div
      className="lumx-user-block__wrapper"
    >
      <span
        className="lumx-user-block__name"
        onClick={[Function]}
        tabIndex={0}
      >
        Emmitt O. Lum
      </span>
    </div>
  </div>,
  <div
    className="lumx-user-block lumx-user-block--orientation-horizontal lumx-user-block--size-m lumx-user-block--theme-light"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="lumx-user-block__avatar"
    >
      <Avatar
        alt="Avatar"
        image="avatar1.jpg"
        onClick={[Function]}
        size="m"
        tabIndex={0}
        theme="light"
      />
    </div>
    <div
      className="lumx-user-block__wrapper"
    >
      <span
        className="lumx-user-block__name"
        onClick={[Function]}
        tabIndex={0}
      >
        Emmitt O. Lum
      </span>
      <div
        className="lumx-user-block__fields"
      >
        <span
          className="lumx-user-block__field"
          key="0"
        >
          Creative developer
        </span>
        <span
          className="lumx-user-block__field"
          key="1"
        >
          Denpasar
        </span>
      </div>
    </div>
  </div>,
  <div
    className="lumx-user-block lumx-user-block--orientation-horizontal lumx-user-block--size-l lumx-user-block--theme-light"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="lumx-user-block__avatar"
    >
      <Avatar
        alt="Avatar"
        image="avatar1.jpg"
        onClick={[Function]}
        size="l"
        tabIndex={0}
        theme="light"
      />
    </div>
    <div
      className="lumx-user-block__wrapper"
    >
      <span
        className="lumx-user-block__name"
        onClick={[Function]}
        tabIndex={0}
      >
        Emmitt O. Lum
      </span>
      <div
        className="lumx-user-block__fields"
      >
        <span
          className="lumx-user-block__field"
          key="0"
        >
          Creative developer
        </span>
        <span
          className="lumx-user-block__field"
          key="1"
        >
          Denpasar
        </span>
      </div>
    </div>
  </div>,
]
`;

exports[`<UserBlock> Snapshots and structure should render story 'WithBadge' 1`] = `
<div
  className="lumx-user-block lumx-user-block--orientation-horizontal lumx-user-block--size-m lumx-user-block--theme-light"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="lumx-user-block__avatar"
  >
    <Avatar
      alt="Avatar"
      badge={
        <Badge
          color="blue"
        >
          <Icon
            icon="M12,17.27L18.18,21L16.54,13.97L22,9.24L14.81,8.62L12,2L9.19,8.62L2,9.24L7.45,13.97L5.82,21L12,17.27Z"
          />
        </Badge>
      }
      image="avatar1.jpg"
      onClick={[Function]}
      size="m"
      tabIndex={0}
      theme="light"
    />
  </div>
  <div
    className="lumx-user-block__wrapper"
  >
    <span
      className="lumx-user-block__name"
      onClick={[Function]}
      tabIndex={0}
    >
      Emmitt O. Lum
    </span>
    <div
      className="lumx-user-block__fields"
    >
      <span
        className="lumx-user-block__field"
        key="0"
      >
        Creative developer
      </span>
      <span
        className="lumx-user-block__field"
        key="1"
      >
        Denpasar
      </span>
    </div>
  </div>
</div>
`;
